hpx/runtime/threads/executors/guided_pool_executor.hpp

```diff
@@ -8,47 +8,53 @@
 #define HPX_RUNTIME_THREADS_GUIDED_POOL_EXECUTOR
 
 #include <hpx/async.hpp>
-#include <hpx/runtime/threads/thread_executor.hpp>
-#include <hpx/runtime/threads/executors/pool_executor.hpp>
-#include <hpx/runtime/threads/thread_pool_base.hpp>
-#include <hpx/lcos/dataflow.hpp>
-#include <hpx/functional/bind_back.hpp>
 #include <hpx/debugging/demangle_helper.hpp>
+#include <hpx/debugging/print.hpp>
+#include <hpx/functional/bind_back.hpp>
 #include <hpx/functional/invoke.hpp>
+#include <hpx/lcos/dataflow.hpp>
+#include <hpx/runtime/threads/executors/pool_executor.hpp>
+#include <hpx/runtime/threads/thread_executor.hpp>
+#include <hpx/runtime/threads/thread_pool_base.hpp>
+#include <hpx/traits/is_future_tuple.hpp>
 #include <hpx/util/pack_traversal.hpp>
 #include <hpx/util/thread_description.hpp>
-#include <hpx/traits/is_future_tuple.hpp>
 
 #include <cstddef>
 #include <cstdint>
 #include <string>
 #include <type_traits>
 #include <utility>
 
-//#define GUIDED_EXECUTOR_DEBUG 1
 //#define GUIDED_POOL_EXECUTOR_FAKE_NOOP
 
-#ifdef GUIDED_EXECUTOR_DEBUG
-#include <iostream>
+#include <hpx/config/warnings_prefix.hpp>
+
+#if !defined(GUIDED_POOL_EXECUTOR_DEBUG)
+#define GUIDED_POOL_EXECUTOR_DEBUG false
 #endif
 
-#include <hpx/config/warnings_prefix.hpp>
+namespace hpx {
+    // cppcheck-suppress ConfigurationNotChecked
+    static hpx::debug::enable_print<GUIDED_POOL_EXECUTOR_DEBUG> gpx_deb(
+        "GP_EXEC");
```

Comment:
Are we sure that the use of the `enable_print` like this does not cause any ODR violations?

Reply:
I have not had any link problems though I can see that if 2 TUs had different values of the macro, it would be a problem. I think it's up to individuals to not do that if they use it